models/keyword_test.go

```diff
@@ -62,6 +64,46 @@ func (s *KeywordDBTestSuite) TestSaveKeywordsWithValidParams() {
 	assert.Equal(s.T(), "Mbappe", result[4].Keyword)
 }
 
+func (s *KeywordDBTestSuite) TestSaveKeywordsWithSingleValidParams() {
```

Comment:
Could you please explain more why save keyword one by one in the transaction is better than save the whole keyword list without transaction? 🙏

Reply:
Mostly because it's less complex and more flexible.

Think of a keyword as a resource with its own lifecycle. Each keyword must be first saved to the database and then a job must be enqueued. Each step must be happening for every new keyword. 

Currently, both steps are not done the same place. That's introducing complexity. There are two loops:
- One loop in `KeywordService` to create a list of a keyword to be saved.
- Second loop in `EnqueueScrapingJob` to enqueue each keyword job.

What I am recommending is to have a single loop in `KeywordService` to:
- Save each keyword
- Enqueue a new job for each keyword right after saving the keyword

Reply:
- Updated SaveKeyword in model in ca9fa05f6358211522b467fcc0d5837d277a6090
- Updated service (without transaction) in e43d3a88a0b761c402ddd9d9e54d3fe4013fe562

Reply:
Added transaction in ee4c0a05658f1af67d3b74d0279325966de88f8d.
It looks weird but this is the only way I figured out.
@olivierobert Do you have any comment? or can we leave this PR without transaction?

Reply:
It's kind of weird indeed as it requires to pass the transaction to the method `SaveKeyword` as an argument:

```go
models.SaveKeyword(keyword, tx)
```

I checked the docs and it seems to be the way to go but there might be better ways to do it. Up to you to keep it or not.

Reply:
> It's kind of weird indeed as it requires to pass the transaction to the method `SaveKeyword` as an argument:
> 
> ```go
> models.SaveKeyword(keyword, tx)
> ```
> 
> I checked the docs and it seems to be the way to go but there might be better ways to do it. Up to you to keep it or not.

I think I would keep it as I think this kind of thing should has a transaction.

Reply:
I don't have a better option as well for now 😔 So it can be marked as a FIXME

Reply:
Added TODO in 1807813c580b9a0df5262c564930e63b3c9b8bd2